### PR TITLE
Scenario flag

### PR DESCRIPTION
**Pull request by bdemers**
scenario flag: 78d2d6d945f083224e2ca297f3a0aa1c27652318
lint: 222b2b187f22a5c72fea8edeb1bc265ff03130f4

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -21,6 +25,7 @@ util.parseCommandLineArgs = defaults => {
       `Directory with the tapes in HAR format. defaults to ${defaults.tapeDir}`,
       util.formatPath, defaults.tapeDir)
     .option('-r, --record', 'Erase all files in tapeDir and record new tapes')
+    .option('-s, --scenarios <items>', 'Comma seperated list of the scenaros to run', util.splitList)
```
Comment (lboyette-okta) [inline lib/util.js]: `comma-separated` `scenarios`

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "fs-extra": "3.0.1",
     "json-stable-stringify": "1.0.1",
     "lodash": "4.17.4",
-    "request": "2.81.0"
+    "request": "2.81.0",
+    "eslint": "^3.19.0"
```
Comment (lboyette-okta) [inline package.json]: we use specific versions, so no caret

File: lib/server.js [hunk 3]
```diff
@@ -134,23 +135,29 @@ server.start = config => {
   // Populate map
   if (app.record) {
     fs.removeSync(app.tapeDir);
+  } else if (app.scenarios && app.scenarios.length > 0) {
+    // Load scenarios from arg list
+    app.scenarios
+        .map(scenario => {
+          let entry = harUtil.loadHarFromFile(path.join(app.tapeDir, `${scenario}.har`));
```
Comment (lboyette-okta) [inline lib/server.js]: I know it's weird, but `let` should be `const` unless it's reassigned somewhere

File: lib/server.js [hunk 4]
```diff
@@ -134,23 +135,29 @@ server.start = config => {
...
-    // Load existing scenarios
+    // Load all scenarios
     fs.readdirSync(app.tapeDir)
       .map(scenarioFileName => {
-        const scenarioName = path.basename(scenarioFileName, '.har');
-        const scenarioFileLoc = path.join(app.tapeDir, scenarioFileName);
-        const scenario = fs.readJsonSync(scenarioFileLoc);
-        app.scenarioEntriesMap[scenarioName] = scenario.log.entries;
+        let entry = harUtil.loadHarFromFile(path.join(app.tapeDir, scenarioFileName));
+        app.scenarioEntriesMap[entry[0]] = entry[1];
```
Comment (lboyette-okta) [inline lib/server.js]: `let entry` => `const [scenarioName, scenarioEntries]`
`app.scenarioEntriesMap[scenarioName] = scenarioEntries;`

File: lib/harUtil.js [hunk 5]
```diff
@@ -101,4 +102,11 @@ harUtil.createHarEntry = ({proxyTarget, responseBody, req, res}) => {
     request,
     response
   };
-}
+};
+
+harUtil.loadHarFromFile = scenarioFileName => {
+  const scenarioName = path.basename(scenarioFileName, '.har');
+  const scenarioFileLoc = scenarioFileName; //path.join(app.tapeDir, scenarioFileName);
+  const scenario = fs.readJsonSync(scenarioFileLoc);
```
Comment (lboyette-okta) [inline lib/harUtil.js]: we can get rid of line 109 and just rename `scenarioFileLoc` to `scenarioFileName`

File: package.json (hunk 2 above)
Comment (bdemers) [inline package.json]: +1 (copied from our node sdk, so we should update that too)

File: lib/server.js (hunk 4 above)
Comment (bdemers) [inline lib/server.js]: thanks, i was wondering what the correct way to handle this was

File: lib/server.js (hunk 3 above)
Comment (bdemers) [inline lib/server.js]: i was wondering about that

File: lib/harUtil.js [hunk 6]
```diff
@@ -73,7 +74,7 @@ harUtil.createHarEntry = ({proxyTarget, responseBody, req, res}) => {
         });
       }
       return aggregate;
-    }(),
+    }),
```
Comment (lboyette-okta) [inline lib/harUtil.js]: this defines a function without executing it. we need a `()` at the end

**Pull request merged**